### PR TITLE
fix has unknown label is not always initialized

### PR DESCRIPTION
**Pull request by helpmefindaname**
fixes https://github.com/flairNLP/flair/issues/3367

### CODE REVIEW

**Comment by alanakbik** [comment]
@helpmefindaname thanks for fixing this! The error seems to occur because there is no valid data point to be predicted on in the test set, and therefore the `has_unknown_label` is never initialized.

When testing with the example script below, this PR does indeed fix this error. However, it throws another error later when the evaluate code tries to access the `classification_report_dict["accuracy"]` field - since there is no valid data point to make a prediction, no accuracy can be computed.

This error can be reproduced with: 

```python
import flair
from flair.data import Corpus
from flair.datasets import ColumnCorpus, WNUT_17
from flair.embeddings import TransformerWordEmbeddings
from flair.models import SpanClassifier
from flair.trainers import ModelTrainer

flair.set_seed(123)

corpus: Corpus = WNUT_17().downsample(0.01)

label_dictionary = corpus.make_label_dictionary("ner", add_unk=True)

embeddings = TransformerWordEmbeddings(
    model="distilbert-base-uncased", layers="-1", subtoken_pooling="first", fine_tune=True, use_context=True
)

tagger = SpanClassifier(embeddings=embeddings, label_dictionary=label_dictionary)

trainer = ModelTrainer(tagger, corpus)

trainer.fine_tune(
    "./results/",
    max_epochs=1,
)
```

**Comment by alanakbik** [comment]
The second error is another issue, merging this now and opening a separate PR for the above error.
